src/code.google.com/p/codesearch/index/mmap_windows.go

```diff
@@ -11,6 +11,29 @@ import (
 	"unsafe"
 )
 
+// An mmapData is mmap'ed read-only data from a file.
+type mmapData struct {
+  f *os.File
+  d []byte
+  o uintptr
+}
+
+func Munmap(o interface{}) error{
+	// This Munmap function is called from read.go like Munmap(o unitptr) and
+	// from write.go like Munmap(o []byte)
+	// For now, only the calls from read.go are handled using the windows equivalent of Munmap
+	// i.e syscall.UnmapViewOfFile(o uintptr)
+	// Calls from write.go get a nil always.
+	// TODO: Find a way to get the mmapData.o field in write.go.
+
+	switch o.(type) {
+		case uintptr:
+			return syscall.UnmapViewOfFile(o.(uintptr))
+		default:
+			return nil	
```

Comment:
I think the right answer here is to have the index writers also hold a proper reference to the whole mmapData and not just the slice taken from the mmap region. Maybe it's time to have mmapData be an io.Closer so we don't have to keep up with what field needs to be passed to Munmap.